plugins/modules/ec2_transit_gateway.py

```diff
@@ -269,20 +254,19 @@ def process(self):
         elif state == "absent":
             self.ensure_tgw_absent(tgw_id, description)
 
-    def wait_for_status(self, wait_timeout, tgw_id, status, skip_deleted=True):
+    def wait_for_status(self, wait_timeout: int, tgw_id: str, status: str, skip_deleted: bool = True) -> Dict[str, Any]:
         """
-        Wait for the Transit Gateway to reach  the specified status
+        Wait for the Transit Gateway to reach the specified status.
         :param wait_timeout: Number of seconds to wait, until this timeout is reached.
         :param tgw_id: The Amazon nat id.
         :param status: The status to wait for.
-                examples. status=available, status=deleted
-        :param skip_deleted: ignore deleted transit gateways
-        :return dict: transit gateway object
+        :param skip_deleted: Ignore deleted transit gateways.
+        :return: Transit gateway object.
         """
         polling_increment_secs = 5
         wait_timeout = time() + wait_timeout
         status_achieved = False
-        transit_gateway = dict()
+        transit_gateway = {}
```

Comment:
Is it possible to create a custom waiter for this instead?

Reply:
Have you updated the code to use the custom waiter instead because I still see the old logic though?

Reply:
in progress to implement

Reply:
implemented custom waiter https://github.com/ansible-collections/amazon.aws/pull/2325